### PR TITLE
Improve comparators

### PR DESCRIPTION
**Pull request by apasel422**

### CODE REVIEW

File: src/compare.rs [hunk 1]
```diff
@@ -229,6 +230,14 @@ impl<F: ?Sized, Lhs: ?Sized, Rhs: ?Sized> Compare<Lhs, Rhs> for F
     fn compare(&self, lhs: &Lhs, rhs: &Rhs) -> Ordering { (*self)(lhs, rhs) }
 }
 
+impl<'a, Lhs: ?Sized, Rhs: ?Sized, C: ?Sized> Compare<Lhs, Rhs> for &'a C
```
Comment (Gankro) [inline src/compare.rs]: Why is this necessary?

Comment (apasel422) [inline src/compare.rs]: It's not. It's just an enhancement that says that you can use a reference to a comparator as a comparator.

**Comment by Gankro** [comment]
LGTM, I guess? Does this remove all the warnings for comparators?

**Comment by Gankro** [comment]
Looks like it does.
